### PR TITLE
Customizing the Platform

### PR DESCRIPTION
**Pull request by munoz0raul**

### CODE REVIEW

**Comment by doanac** [comment]
Docs for cb5f60af02d2dc0bd52e3d28c983fac30bfc9382 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/923/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 4eddb40b9bc35e8c7fa4554ab48a101099d90661 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/924/docs/artifacts/html/index.html

File: source/tutorials/customizing-the-platform/customizing-the-platform.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+Customizing the Platform
+========================
+
+All the previous tutorials covered the application development with Docker Compose Application.
+
+Docker Compose Apps are the recommended way to create applications but you are 
+also allowed to customize and change the platform according to your needs.
+
+Customizations in the platform usually are necessary when your need to add new
```
Comment (vanmaegima) [inline source/tutorials/customizing-the-platform/customizing-the-platform.rst:11]: Customizations in the platform usually are necessary when **you** need to add new

File: source/tutorials/customizing-the-platform/customizing-the-platform.rst [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+to change anything on the platform.
+
+This tutorial will guide you over simple ``meta-subscriber-overrides`` customization. 
+The same application used in the previous tutorial, the ``shellhttpd``, will 
+be added to your platform and will start during boot.
+
+.. tip::
+
+  If you are not familiar with Yocto/OpenEmbedded you can still follow this tutorial. 
+  FoundriesFactory CI will help makeing it simple by building the platform in the cloud.
```
Comment (vanmaegima) [inline source/tutorials/customizing-the-platform/customizing-the-platform.rst:41]: making

File: source/tutorials/customizing-the-platform/cloning-meta-sub-repository.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
+Cloning Meta Subscriber Overrides Repository
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. tip::
+
+   When your Factory is first created, 2 branches are established: ``master`` and ``devel``.
```
Comment (vanmaegima) [inline source/tutorials/customizing-the-platform/cloning-meta-sub-repository.rst:5]: 2 branches are established for meta-subscriber-overrides

**Review by vanmaegima — APPROVED** [review]

File: source/tutorials/customizing-the-platform/testing-recipe.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+.. prompt:: bash device:~$, auto
+
+    device:~$  systemctl status shellhttpd
+
+**Example Output**:
+
+.. prompt:: text
+
+    ● shellhttpd.service - Start up Shellhttpd Application
```
Comment (angolini) [inline source/tutorials/customizing-the-platform/testing-recipe.rst:31]: ● 
Could you, please, double check this character?

File: source/tutorials/customizing-the-platform/testing-recipe.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+         Loaded: loaded (/usr/lib/systemd/system/shellhttpd.service; enabled; vendor preset: enabled)
+         Active: active (running) since
+       Main PID: 404 (sh)
+          Tasks: 2 (limit: 777)
+         Memory: 2.1M
+         CGroup: /system.slice/shellhttpd.service
+                 ├─404 /bin/sh /usr/share/shellhttpd/httpd.sh
+                 └─413 nc -c -l -p 8080
+
+Using a second terminal, test your applications using ``curl`` in any external
```
Comment (angolini) [inline source/tutorials/customizing-the-platform/testing-recipe.rst:41]:
```suggestion
Using a second terminal, test your application using ``curl`` in any external 
```

File: source/tutorials/customizing-the-platform/testing-recipe.rst [hunk 6]
```diff
@@ -0,0 +1,66 @@
+Testing Applications
+^^^^^^^^^^^^^^^^^^^^
+
+Platform changes create an incremental OSTree change. In order to apply platform 
+updates, the device needs to reboot.
```
Comment (angolini) [inline source/tutorials/customizing-the-platform/testing-recipe.rst:5]: Maybe include this "OSTree" information in this tutorial is too complex.
I think line 4 and 5 can be excluded.
The link 7 is talking about the need of a reboot, so I think it's ok

**Review by angolini — CHANGES_REQUESTED** [review]
I've added suggestions.
The only thing that I think it's critical to change is when using "Yocto" instead of "the Yocto Project"

**Comment by doanac** [comment]
Docs for b1c9d0f0f7eab26ad4b9b6fe18920295d2e037c2 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/927/docs/artifacts/html/index.html

File: source/tutorials/customizing-the-platform/customizing-the-platform.rst [hunk 7]
```diff
@@ -0,0 +1,74 @@
+.. _tutorial-customizing-the-platform:
+
+Customizing the Platform
+========================
+
+All the previous tutorials covered the application development with Docker Compose Application.
```
[the remark this answers is not in the document]
Comment (munoz0raul) [reply source/tutorials/customizing-the-platform/customizing-the-platform.rst:6]: I will remove this line, It is not necessary!

File: source/tutorials/customizing-the-platform/customizing-the-platform.rst [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+Docker Compose Apps are the recommended way to create applications but you are 
+also allowed to customize and change the platform according to your needs.
```
[the remark this answers is not in the document]
Comment (munoz0raul) [reply source/tutorials/customizing-the-platform/customizing-the-platform.rst:9]: I will wait for Mike review. He will probably change this line. In my sentence, I tried to cover any customization. On your suggestion, you are covering just adding the app. So, let's way Mike and Tim here.

File: source/tutorials/customizing-the-platform/customizing-the-platform.rst [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+Customizations in the platform usually are necessary when your need to add new 
+packages to the platform, customize the Linux for specific hardware requirements, 
+update the Linux microPlatform, and so on.
+
+Your Factory platform is a derivative of the Foundries.io Linux microPlatform. 
+Foundries.io is responsible for maintaining the Linux microPlatform for the 
+reference hardwares and you have the total control to customize and change your platform.
```
[the remark this answers is not in the document]
Comment (munoz0raul) [reply source/tutorials/customizing-the-platform/customizing-the-platform.rst:17]: Changed it!

File: source/tutorials/customizing-the-platform/customizing-the-platform.rst [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+Foundries.io updates in the Linux microPlatform also don't automatically apply
```
[the remark this answers is not in the document]
Comment (munoz0raul) [reply source/tutorials/customizing-the-platform/customizing-the-platform.rst:19]: Changed!

**Comment by doanac** [comment]
Docs for b890cd68deb5c95096f268e30d18b076ea27ff52 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/928/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 0e81312361df6a8ddd6d3f7c2dc486f49e6da437 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/935/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 3043357d89a0685745c9ce02ac95008ac95e8f25 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1004/docs/artifacts/html/index.html

**Review by angolini — APPROVED** [review]

File: source/tutorials/customizing-the-platform/customizing-the-platform.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (munoz0raul) [reply source/tutorials/customizing-the-platform/customizing-the-platform.rst:7]: @Tim-Anderson could you take a look on this part as well.

**Comment by doanac** [comment]
Docs for 79939cdb478a3a4bdecacbbaeb3493b2c656d3cb are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1036/docs/artifacts/html/index.html

File: source/tutorials/customizing-the-platform/customizing-the-platform.rst [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+========================
+
+Docker Compose Apps are the recommended way to create applications but you are 
+also allowed to customize and change the platform according to your needs.
+
+Customization in the platform is usually useful when you need to add 
+packages to the platform, customize the Linux for specific hardware requirements, 
+update the Linux microPlatform, and so on.
+
+Foundries.io frequently update the Linux microPlatform for the reference hardwares;
```
Comment (Tim-Anderson) [inline source/tutorials/customizing-the-platform/customizing-the-platform.rst:13]: updates

File: source/tutorials/customizing-the-platform/customizing-the-platform.rst [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+The ``lmp-manifest.git`` contains a manifest file which gather the meta-layers used to build 
+your image. This is the repository you should change to update your platform to newer 
+Linux microPlatform versions. For more information, read :ref:`ref-linux-update`.
+
+The ``meta-subscriber-overrides`` is the suggested meta-layer for  
+customization. That layer is a high-priority layer and it gives you the power 
+to change anything on the platform.
+
+This tutorial guides you over simple ``meta-subscriber-overrides`` customization.
```
Comment (Tim-Anderson) [inline source/tutorials/customizing-the-platform/customizing-the-platform.rst:28]: guides you through

File: source/tutorials/customizing-the-platform/shellhttpd-recipe.rst [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+
+    mkdir recipes-support
+
+In the ``recipes-support`` folder, use git to download the ``shellhttpd`` recipe 
+from the extra-meta-subscriber-overrides repository:
+
+.. prompt:: bash host:~$, auto
+
+    host:~$ cd recipes-support
+    host:~$ git remote add fio https://github.com/munoz0raul/extra-meta-subscriber-overrides.git
```
Comment (Tim-Anderson) [inline source/tutorials/customizing-the-platform/shellhttpd-recipe.rst:22]: is the repository in your private path munoz0raul ?

**Review by Tim-Anderson — COMMENTED** [review]
correct my comments and it should be good to go

**Comment by doanac** [comment]
Docs for a9822d371b45eeb40611d1570e18930e5f8474d8 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1075/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 42fbdcfb2f01f2fc39cca85470f37e3f2eb81a3f are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1076/docs/artifacts/html/index.html

**Pull request merged**